src/classlibnative/bcltype/number.cpp

```diff
@@ -134,8 +135,191 @@ unsigned int Int64DivMod1E9(unsigned __int64* value)
 
 #else // _TARGET_X86_ && !FEATURE_PAL
 
-#pragma warning(disable:4273)
-extern "C" char* __cdecl _ecvt(double, int, int*, int*);
+void ecvt( double value, int count, int* dec, int* sign, wchar_t* digits )
```

Comment:
If I understand correctly, the `FEATURE_PAL` can indicate whether the OS is Windows or not. If that is correct, I think ifdef out the code like this should resolve the issue:

```cpp
#ifndef  FEATURE_PAL
#pragma warning(disable:4273)
extern "C" char* __cdecl _ecvt(double, int, int*, int*);
#else
void ecvt( double value, int count, int* dec, int* sign, wchar_t* digits )
...
...
#endif
```

And on [line 336](https://github.com/dotnet/coreclr/pull/12894/files/b617b7928f47263980752318951663b7faa79dee#diff-0d0ac521b3af6ae20f3148fa1bee21f1L152):

```cpp
#ifndef FEATURE_PAL
 char* src = _ecvt(value, precision, &number->scale, &number->sign);
 wchar* dst = number->digits;
 if (*src != '0') {
      while (*src) *dst++ = *src++;
 }
 *dst = 0;
#else
ecvt(value, precision, &number->scale, &number->sign, number->digits);
#endif
```